### PR TITLE
[ResourceBundle] Enable max depth check on API serialization

### PR DESCRIPTION
**Pull request by KasraKhosravi**
Q | A
------------- | -------------
Bug fix? | yes
New feature? | no
BC breaks? | no
Deprecations? | no
Fixed tickets | no
License | MIT
Doc PR | -

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
@KasraKhosravi Could you just quickly explain this change? I do not know this method of serializer. Thanks!

File: src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php [hunk 1]
```diff
@@ -166,6 +168,8 @@ public function indexAction(Request $request)
             ->setData($resources)
         ;
 
+        $view->getSerializationContext()->enableMaxDepthChecks();
```
Comment (stloyd) [inline src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php]: This should be done in `handleView()` same as there is group setting being made.

Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php]: :+1:

**Comment by KasraKhosravi** [comment]
@pjedrzejewski Currently using `max_depth` in serializer yaml files does not work because `maxDepth` exclusion strategy is not enabled by default.

You can visit bottom of [this page](http://jmsyst.com/libs/serializer/master/cookbook/exclusion_strategies) in the official docs.

**Comment by josx** [comment]
:+1: 

We need that for our serialization api.

**Comment by pjedrzejewski** [comment]
OK, clear! Good to go as soon as Travis turns green. :)

**Comment by aramalipoor** [comment]
:+1:

**Comment by kayue** [comment]
Didn't know this is something we have to enabled explicitly :+1:

**Comment by jimbocoder** [comment]
Well that certainly explains a few things for me.  Thanks @KasraKhosravi!

**Comment by ahmadrabie** [comment]
:+1:

**Comment by pjedrzejewski** [comment]
Thank you Kasra! :+1:

**Pull request merged**